### DISCUSSION TITLE
'idle' state (removal) in v4 - queries vs. mutations

### DISCUSSION THREAD

**Discussion by targumon** [TanStack/query#4633] [General] [answered]
First of all, thanks for the [great v4 migration guide](https://tanstack.com/query/v4/docs/guides/migrating-to-react-query-4)! 🙏

One of the things I liked is the **streamlining** of Queries & Mutations (e.g. auto garbage collection for mutations just like for queries, or aligning their NotifyEvents).

Which is why I was surprised to see their statuses **diverge**:

`export type QueryStatus = 'loading' | 'error' | 'success'`
`export type MutationStatus = 'loading' | 'error' | 'success' | 'idle'`

I totally understand why 'idle' was removed from Queries.
But now that I have to fix a small bit of my code (because I assumed the types are compatible and they no longer are - it's a small fix, no hassle), I was just wondering if it wasn't supposed to be removed from Mutations as well. 🤔

WDYT?

**Comment by TkDodo** [comment, marked as answer]
queries and mutations are fundamentally different. mutations don't have a `fetchStatus` like queries do. Every mutation starts off in `idle` state, while only a small amount of queries did actually start out in `idle` state.

When we introduced the `fetchStatus` for `queries`, the `idle` state stood out because it couldn't be combined with certain fetch states. For example, you couldn't be `idle` and `paused` - that didn't make sense.

So while unification is generally good, I think we have two different cases here that justify the differences.
